README.md

```diff
@@ -41,38 +42,104 @@ have to explicitly type the property.
 
 ```python
 from django.db import connection, models
-from django.db.models.manager import Manager
+
 
 class User(models.Model):
-    title = models.CharField(max_length=255)
+    objects = models.Manager["User"]()
 
-    objects = Manager["User"]()
 
 reveal_type(User.objects.all().first())
 # note: Revealed type is 'Optional[User]'
 ```
 
-### ForeignKey ids as properties in ORM models
+### ForeignKey ids and related names as properties in ORM models
 
 When defining a Django ORM model with a foreign key, like so:
 
 ```python
 class User(models.Model):
-    team = models.ForeignKey("Team", null=True, on_delete=models.SET_NULL)
+    team = models.ForeignKey(
+        "Team",
+        null=True,
+        on_delete=models.SET_NULL,
+    )
+    role = models.ForeignKey(
+        "Role",
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="users",
+    )
 ```
 
-two properties are created, `team` as expected, and `team_id`. In order for
-mypy to know about the id property we need to define it manually as follows:
+two properties are created, `team` as expected, and `team_id`. Also, a related
+manager called `user_set` is created on `Team` for the reverse access.
+
+In order to properly add typing to the foreing key itself and also for the created ids you can do
+something like this:
 
 ```python
 from typing import TYPE_CHECKING
 
+from someapp.models import Team
+if TYPE_CHECKING:
+    # In this example Role cannot be imported due to circular import issues,
+    # but doing so inside TYPE_CHECKING will make sure that the typing bellow
+    # knows what "Role" means
+    from anotherapp.models import Role
+
+
 class User(models.Model):
-    team = models.ForeignKey("Team", null=True, on_delete=models.SET_NULL)
+    team_id: Optional[int]
+    team = models.ForeignKey[Team](
```

Comment:
I think if we pass in `Team` as the first param we shouldn't need the explicit type param:

```suggestion
    team = models.ForeignKey(
```